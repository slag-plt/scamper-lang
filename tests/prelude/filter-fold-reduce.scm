(filter string? (list 4 "HelloWorld" "HelloWorld" "HelloWorld" "HelloWorld" #t "HelloWorld" "HelloWorld" "HelloWorld" list))
(filter procedure? (list string-length list? + - "true" #f = string?))
(filter null? (list (pair 2 4) (pair "a" "b") (pair "first" "second") (list) (list 4 5 6)))
(filter list? (list (pair 2 4) (pair "a" "b") (pair "first" "second") (list) (list 4 5 6)))
(fold + 5 (list 0 1 2 3 4))
(fold - 5 (list 10 9 8 7 6))
(reduce + (list 5 0 1 2 3 4))
(reduce - (list 5 10 9 8 7 6))