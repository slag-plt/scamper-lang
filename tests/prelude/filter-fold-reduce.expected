(list "HelloWorld" "HelloWorld" "HelloWorld" "HelloWorld" "HelloWorld" "HelloWorld" "HelloWorld")
(list [prim stringLengthPrim] [prim listQPrim] [prim plusPrim] [prim minusPrim] [prim numeqPrim] [prim stringQPrim])
(list null)
(list null (list 4 5 6))
15
-35
15
-35
